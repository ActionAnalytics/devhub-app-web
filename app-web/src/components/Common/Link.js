import React from 'react';
import Link from 'gatsby-link';

const GatsbyLink = ({ children, to, ...rest }) => {
  // Tailor the following test to your environment.
  // This example assumes that any internal link (intended for Gatsby)
  // will start with exactly one slash, and that anything else is external.
  const internal = /^\/(?!\/)/.test(to);

  // Use Gatsby Link for internal links, and <a> for rests
  if (internal) {
    return (
      <Link to={to} {...rest}>
        {children}
      </Link>
    );
  }
  return (
    <a href={to} {...rest}>
      {children}
    </a>
  );
};

export default GatsbyLink;
