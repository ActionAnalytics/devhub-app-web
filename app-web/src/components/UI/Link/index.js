import Link from './Link';
import SearchAwareLink from './SearchAwareLink';

export { Link, SearchAwareLink };
