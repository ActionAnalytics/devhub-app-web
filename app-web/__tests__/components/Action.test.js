import React from 'react';
import { shallow } from 'enzyme';
import Action from '../../src/components/Cards/Backside/ActionGroup/Action/Action';
import Link from '../../src/components/UI/Link/Link';
describe('Action Component (card)', () => {
  it('matches snapshot', () => {
    const props = {
      icon: <i>Icon</i>,
      actionName: 'foo',
      actionValue: 'bar',
    };

    const wrapper = shallow(<Action {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a link when link prop is passed in', () => {
    const props = {
      icon: <i>Icon</i>,
      actionName: 'foo',
      actionValue: 'bar',
      link: '/something',
    };

    const wrapper = shallow(<Action {...props} />);
    const anchorTag = wrapper.find(Link).first();
    expect(anchorTag.props().to).toBe(props.link);
  });

  it('renders no link when link prop is NOT passed in', () => {
    const props = {
      icon: <i>Icon</i>,
      actionName: 'foo',
      actionValue: 'bar',
    };

    const wrapper = shallow(<Action {...props} />);
    const anchorTag = wrapper.find(Link);
    expect(anchorTag.exists()).toBe(false);
  });
});
