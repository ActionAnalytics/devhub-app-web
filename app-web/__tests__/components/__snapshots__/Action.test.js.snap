// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Component (card) matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Action
    actionName="foo"
    actionValue="bar"
    icon={
      <i>
        Icon
      </i>
    }
    link=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Aux>
        <i>
          Icon
        </i>
        <span>
          bar
        </span>
      </Aux>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <i>
            Icon
          </i>,
          <span>
            bar
          </span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Icon",
          },
          "ref": null,
          "rendered": "Icon",
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "bar",
          },
          "ref": null,
          "rendered": "bar",
          "type": "span",
        },
      ],
      "type": [Function],
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Aux>
          <i>
            Icon
          </i>
          <span>
            bar
          </span>
        </Aux>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <i>
              Icon
            </i>,
            <span>
              bar
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Icon",
            },
            "ref": null,
            "rendered": "Icon",
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "bar",
            },
            "ref": null,
            "rendered": "bar",
            "type": "span",
          },
        ],
        "type": [Function],
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
